client/src/components/Search/Search.tsx

```diff
@@ -1,46 +1,59 @@
-import React, { FunctionComponent, useState, useRef, ChangeEvent } from 'react';
+import React, {
+  FunctionComponent,
+  useEffect,
+  useState,
+  useRef,
+  useCallback,
+  ChangeEvent
+} from 'react';
+import { useDispatch } from 'react-redux';
 
 // Store
-import { useAppSelector } from '../../../app/hooks';
-
-// API
-import { useSearchLocationsQuery } from '../../../services/forecaApi';
+import { useAppSelector } from 'redux/hooks';
+import { searchLocations } from 'redux/actionCreators/location';
 
 // Custom hooks
-import useDebounce from '../../hooks/useDebounce';
-import useOnClickOutside from '../../hooks/useOnClickOutside';
+import useDebounce from 'hooks/useDebounce';
+import useOnClickOutside from 'hooks/useOnClickOutside';
+
+// Assets
+import { IconSearch } from 'assets/images/svg';
 
 // Components
-import Input from '../Input';
+import Input from '../Input/Input';
 import SearchResultList from './SearchResultList';
 
 // Styles
-import * as S from './styles';
-
-// Assets
-import { IconSearch } from '../../assets/images/svg';
+import * as S from './Search.styles';
 
 const Search: FunctionComponent = () => {
   // State
   const [searchQuery, setSearchQuery] = useState<string>('');
   const [listShown, setListShown] = useState<boolean>(true);
   const debouncedSearchQuery = useDebounce(searchQuery, 500);
-  const { data } = useSearchLocationsQuery(debouncedSearchQuery, {
-    skip: debouncedSearchQuery === ''
-  });
-  const { theme } = useAppSelector(state => state.theme);
+  const theme = useAppSelector(state => state.theme.theme);
+  const { data } = useAppSelector(state => state.location.search);
 
   const searchRef = useRef<HTMLDivElement>(null);
 
+  const dispatch = useDispatch();
+
   // Handlers
   const handleInputChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
     setSearchQuery(e.currentTarget.value);
   };
   const handleInputFocus = (): void => setListShown(true);
   const handleClickOutside = (): void => setListShown(false);
+  const handleSearchLocations = useCallback(() => {
+    if (debouncedSearchQuery !== '') dispatch(searchLocations(debouncedSearchQuery));
```

Comment:
please make condition a bit shorter, no need to compare exactly with empty string :)